src/datepickers/datepicker.ts

```diff
@@ -149,7 +149,9 @@ export class NglDatepicker {
 
   private daysInPreviousMonth(year: number, month: number) {
     const first = new Date(year, month, 1);
-    const offset = first.getDay();
+    let offset = first.getDay();
+    if (offset === 0 && this.firstDayOfWeek !== 0)
+      offset = 7;
```

Comment:
@sdiprizio Why `7` here? Doesn't this depend on the value of `this.firstDayOfWeek`?